### PR TITLE
Implement tokenIdERC1155 module to handle tokenId management

### PR DESCRIPTION
**Pull request by GWSzeto**

### CODE REVIEW

File: src/core/token/ERC1155Core.sol [hunk 1]
```diff
@@ -173,6 +174,10 @@ contract ERC1155Core is ERC1155, Core, Multicallable, EIP712 {
             selector: UpdateMetadataCallbackERC1155.updateMetadataERC1155.selector,
             mode: CallbackMode.REQUIRED
         });
+        supportedCallbackFunctions[8] = SupportedCallbackFunction({
+            selector: UpdateTokenIdCallbackERC1155.updateTokenIdERC1155.selector,
+            mode: CallbackMode.REQUIRED
```
Comment (joaquim-verges) [inline src/core/token/ERC1155Core.sol:179]: would it make sense to have it optional, and by default just return the token id passed in?

File: src/module/token/tokenId/tokenId1155.sol [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    function data() internal pure returns (Data storage data_) {
+        bytes32 position = TOKEN_ID_STORAGE_POSITION;
+        assembly {
+            data_.slot := position
+        }
+    }
+
+}
+
+contract TokenIdERC1155 is Module, UpdateTokenIdCallbackERC1155 {
```
Comment (joaquim-verges) [inline src/module/token/tokenId/tokenId1155.sol:30]: would call this `SequentialTokenIdERC1155`

File: src/core/token/ERC1155Core.sol (hunk 1 above)
Comment (GWSzeto) [reply src/core/token/ERC1155Core.sol]: Makes sense to me, updated

File: src/module/token/tokenId/tokenId1155.sol (hunk 2 above)
Comment (GWSzeto) [reply src/module/token/tokenId/tokenId1155.sol]: updated

**Review by kumaryash90 — APPROVED** [review]

**Pull request merged**